tests/taskcluster/lib/decorate_task.js

```diff
@@ -96,12 +96,33 @@ function decorateTask(task, options) {
   output.task.scopes.push('docker-worker:image:' + IMAGE + '*');
 
   if (process.env.TREEHERDER_PROJECT && process.env.TREEHERDER_REVISION) {
+    // defaults to 'tc-treeherder' only so existing tasks need not be changed
+    var prefixes = ['tc-treeherder.'];
+
+    if (output.task['extra']) {
+      if (output.task.extra['treeherderEnv']) {
+        prefixes = [];
+        if (output.task.extra['treeherderEnv'].indexOf('staging') != -1) {
+          prefixes.push('tc-treeherder-stage.');
+        }
+        if (output.task.extra['treeherderEnv'].indexOf('production') != -1) {
+          prefixes.push('tc-treeherder.');
+        }
+        // if treeherderEnv provided but not 'staging' or 'production', default to staging
+        if (prefixes.length == 0) {
+          prefixes.push('tc-treeherder-stage.');
+        }
+      }
+    }
+
     output.task.routes = output.task.routes || [];
-    output.task.routes.push(
-      'tc-treeherder.' +
-      process.env.TREEHERDER_PROJECT + '.' +
-      process.env.TREEHERDER_REVISION
-    );
+    prefixes.forEach(function(prefix) {
+      output.task.routes.push(
+        prefix +
```

Comment:
Not sure if it helped readability, but maybe there is a way to do this without concatenating strings especially when there is a separator (dot) in this case.

Some suggestions:
[prefix, process.env.TREEHERDER_PROJECT, process.env.TREEHERDER_REVISION].join('.')

var util = require('util');
var treeherderRoute = util.format('%s.%s.%s', prefix, process.env.TREEHERDER_PROJECT, process.env.TREEHERDER_REVISION)

I know we couldn't use it here, but doing things like this will make it easier to transition to some string formatting goodness in es6:
`${prefix}.${process.env.TREEHERDER_PROJECT}.${process.env.TREEHERDER_REVISION}`